deepset_cloud_sdk/workflows/pipeline_client/models.py

```diff
@@ -0,0 +1,125 @@
+"""Models for the pipeline service."""
+from typing import List
+
+from pydantic import BaseModel, Field, model_validator
+
+
+class PipelineInputs(BaseModel):
+    """Input configuration for the pipeline.
+
+    :param query: List of component names that will receive the query input
+        Each component should be specified as '<component-name>.<run-parameter-name>', e.g., 'retriever.query'
+    :param filters: Optional list of component names that will receive filters input.
+        Each component should be specified as '<component-name>.<run-parameter-name>', e.g., 'retriever.filters'
+    """
+
+    model_config = {"extra": "allow"}  # Allow additional fields in inputs
+
+    query: List[str] = Field(
+        ...,
+        description=(
+            "List of component names that will receive the query input when being executed. "
+            "Format: '<component-name>.<run-parameter-name>', e.g., 'retriever.query'"
+        ),
+        min_items=1,
+    )
+    filters: List[str] = Field(
+        default_factory=list,
+        description=(
+            "List of component names that will receive filters input when being executed. "
+            "Format: '<component-name>.<run-parameter-name>', e.g., 'retriever.filters'"
+        ),
+    )
+
+
+class PipelineOutputs(BaseModel):
+    """Output configuration for the pipeline.
+
+    Must define at least one of `documents` or `answers`, or both.
+
+    :param documents: Component name that will provide documents as output.
+        Should be specified as '<component-name>.<output-parameter>', e.g., 'retriever.documents'
+    :param answers: Component name that will provide answers as output.
+        Should be specified as '<component-name>.<output-parameter>', e.g., 'reader.answers'
+    """
+
+    model_config = {"extra": "allow"}  # Allow additional fields in outputs
+
+    documents: str | None = Field(
+        default=None,
+        description="Component name that will provide documents as output. "
+        "Format: '<component-name>.<output-parameter>', e.g., 'meta_ranker.documents'",
+    )
+    answers: str | None = Field(
+        default=None,
+        description="Component name that will provide answers as output. "
+        "Format: '<component-name>.<output-parameter>', e.g., 'answers_builder.answers'",
+    )
+
+    @model_validator(mode="after")
+    def validate_documents_xor_answers(self) -> "PipelineOutputs":
```

Comment:
```suggestion
    def validate_documents_xor_answers(self) -> PipelineOutputs:
```

+ from __future__ import annoations ?